sql/2023/01/no-scripts-on-page.sql

```diff
@@ -0,0 +1,33 @@
+# HTTP Archive query to get % of WordPress sites with no scripts at all on the page.
+#
+# WPP Research, Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SELECT
+  har_pages._TABLE_SUFFIX AS client,
+  COUNTIF(CAST(JSON_EXTRACT(JSON_EXTRACT_SCALAR(payload, '$._javascript'), '$.script_tags.total') AS INT64) = 0) / COUNT(0) AS total_pct,
```

Comment:
This query gets the number of sites with no `script` tags at all on the page, which may be useful in theory. That includes inline scripts. But since we are researching the usage of `defer`, and inline scripts cannot be deferred anyway, I think we are looking for sites with no _external_ `script` tags actually.

Similarly to the above, https://github.com/GoogleChromeLabs/wpp-research/pull/29/files#diff-d647443f186ed55d494b9389bc8322970137af7a407a69bcc8109c370cff0001 already includes the number of sites with _any_ external scripts, so that means we already have the metric we need (`total_wp_sites - with_any_external_scripts = without_any_external_scripts`). So I think we can remove this query as well.